test/parallel/test-os.js

```diff
@@ -101,3 +102,15 @@ switch (platform) {
 
 var EOL = os.EOL;
 assert.ok(EOL.length > 0);
+
+
+var home = os.homedir();
+
+console.log('homedir = ' + home);
+assert.ok(home.length > 0);
+
+if (isWindows && process.env.USERPROFILE) {
+  assert.equal(home, process.env.USERPROFILE);
+} else if (!isWindows && process.env.HOME) {
+  assert.equal(home, process.env.HOME);
+}
```

Comment:
@bnoordhuis updated based on your feedback. I added a check that modifies the existing environment. I also added a new test that spawns a child process without the appropriate environment variables.